clouddq/classes/dq_configs_cache.py

```diff
@@ -486,17 +473,81 @@ def _resolve_bigquery_entity_uri(
+            logger.info(
                 "Failed to retrieve Dataplex Metadata entry for "
                 f"entity_uri '{entity_uri.complete_uri_string}' "
                 f"with error:\n"
-                f"{pformat(dataplex_entities_match.json())}\n\n"
+                f"{pformat(json.dumps(dataplex_entities_match))}\n\n"
                 f"Parsed entity_uri configs:\n"
                 f"{pformat(entity_uri.to_dict())}\n\n"
             )
-        dataplex_entity = dataplex_entities_match[0]
-        clouddq_entity = dq_entity.DqEntity.from_dataplex_entity(
-            entity_id=entity_uri.get_db_primary_key(),
-            dataplex_entity=dataplex_entity,
-        )
+            return False
+        else:
+            dataplex_entity = dataplex_entities_match[0]
+            clouddq_entity = dq_entity.DqEntity.from_dataplex_entity(
+                entity_id=entity_uri.get_db_primary_key(),
+                dataplex_entity=dataplex_entity,
+            )
         return clouddq_entity
+
+    def _resolve_bigquery_entity_uri(
+        self,
+        entity_uri: dq_entity_uri.EntityUri,
+        client: clouddq_dataplex.CloudDqDataplexClient,
+        bigquery_client: BigQueryClient,
+        enable_experimental_bigquery_entity_uris: bool = True,
+    ) -> dq_entity.DqEntity:
+        if not enable_experimental_bigquery_entity_uris:
+            raise NotImplementedError(
+                f"entity_uri '{entity_uri.complete_uri_string}' "
+                "has unsupported scheme 'bigquery://'.\n"
+                "Use CLI flag --enable_experimental_bigquery_entity_uris "
+                "to enable looking up bigquery:// entity_uri scheme in format "
+                "bigquery://projects/<project-id>/datasets/<dataset-id>/tables/<table-id> "
+                "schemes using Dataplex Metadata API.\n"
+                "Ensure the BigQuery dataset containing this table "
+                "is registered as an asset in Dataplex.\n"
+                "You can then specify the corresponding Dataplex "
+                "projects/locations/lakes/zones as part of the "
+                "metadata_default_registries YAML configs, e.g.\n"
+                f"{SAMPLE_DEFAULT_REGISTRIES_YAML}"
+            )
+        required_arguments = ["projects", "datasets", "tables"]
+        for argument in required_arguments:
+            uri_argument = entity_uri.get_configs(argument)
+            if not uri_argument:
+                raise RuntimeError(
+                    f"Failed to retrieve default Bigquery '{argument}' for "
+                    f"entity_uri: {entity_uri.complete_uri_string}. \n"
+                    f"'{argument}' is a required argument to look-up metadata for the entity_uri "
+                    "using Bigquery API.\n"
+                )
+
+        project_id = entity_uri.get_configs("projects")
+        table_name = entity_uri.get_table_name()
+        bq_native_table_exists = bigquery_client.is_table_exists(
```

Comment:
Is this check for the BigQuery native tables only, or BigQuery native tables and external tables? If the latter, then maybe just call it `bq_table_exists`?

On a related note, it's probably good to clarify what "Bigquery Native Tables Support" entails exactly (regarding external table support).

Reply:
It can be native or external table created by dataplex. We can rename it to the bq_table_exists.